cswinrt/strings/additions/Windows.Storage.Streams/WindowsRuntimeBufferExtensions.cs

```diff
@@ -85,7 +85,7 @@ public static void CopyTo(this byte[] source, int sourceIndex, IBuffer destinati
             if (destination == null) throw new ArgumentNullException(nameof(destination));
             if (count < 0) throw new ArgumentOutOfRangeException(nameof(count));
             if (sourceIndex < 0) throw new ArgumentOutOfRangeException(nameof(sourceIndex));
-            if (source.Length <= sourceIndex) throw new ArgumentException(global::Windows.Storage.Streams.SR.Argument_IndexOutOfArrayBounds, nameof(sourceIndex));
+            if (source.Length < sourceIndex) throw new ArgumentException(global::Windows.Storage.Streams.SR.Argument_IndexOutOfArrayBounds, nameof(sourceIndex));
```

Comment:
I actually think this should be reverted -- we don't want to try and index into the array at it's length value, since arrays are 0-based.

Reply:
No the check is correctly modified, compare with my examples in the PR comment, for example `Array.Copy` allows this to go through. The next two following lines check the upper bound again and prevent overflows, what you check here is that `source.Length - sourceIndex` doesn't overflow, otherwise this check wouldn't be needed at all.

Reply:
Yup, you can get past the safety checks.

```csharp
var array = new byte[3];
var buffer = WindowsRuntimeBuffer.Create(1);
array.CopyTo(0, buffer, 0xFFFFFFFE, 2);
```

This fails in `System.Buffer.BlockCopy` because it casts to `int`, but the `else` branch uses `Marshal.Copy` and will corrupt memory I believe.